### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
- Fix RegEx logic for form validation.
- Add form inputs to local storage.
- Add logic to prefill inputs if there is any data in the local storage.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -161,11 +161,38 @@ const mail = document.getElementById('mail');
...
-  if (mail.value === mail.value.match(RegExp)) {
+  const Regex = /^[a-z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-z0-9-]+(?:\.[a-z0-9-]+)*$/;
+  if (mail.value.match(Regex)) {
     errorMessage.textContent = '';
   } else {
     e.preventDefault();
     errorMessage.innerHTML = '** FORM NOT SENT <br> ** Invalid Email Address ';
   }
 });
+
```
Comment (SajjadAhmad14) [inline main.js:178]: Please make sure that entered values are saved in the browser after filling the form, and not submitting it.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi, 
 ### You both have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
GitHub flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: main.js [hunk 2]
```diff
@@ -161,11 +167,28 @@ const mail = document.getElementById('mail');
 const errorMessage = document.querySelector('.error-message');
 
 formdata.addEventListener('submit', (e) => {
```
Comment (Mihndim2020) [inline main.js:169]: - [ ] As requested by the previous code reviewer, please make sure you entered values are saved to the local storage when filling out the form, not when the form is submitted. Please this is a requirement. You should add a <'change'> event listener to the input fields not a <'submit'> event event listener. Please check out this [article](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event). When you start filling the form and you reload the page, the data you initially entered should  be available in the input fields.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @Chimwemwe-127001, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there


 # Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 2 above)
Comment (Chimwemwe-127001) [reply main.js:169]: - @Mihndim2020, Thank you very much for the clarification. We will go ahead and make the required changes and submit them for the last attempt. Hopefully, everything checks out.

**Review by DanteAlonsoHT — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @Chimwemwe-127001, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**